milestones/1-08-01-2021/tx-indexer/milestone.md

```diff
@@ -0,0 +1,168 @@
+### Milestone Title: Tx Indexer 
+### Leader  
+Eduardo Diaz Blockchain Engineer
+### Date(s)  
+01/14/2021
+### Category  
+- [ ] Infrastructure  
+- [X] Protocol/Blockchain Dev  
+- [ ] App Solutions  
+- [ ] Quality Assurance  
+### Contract  
+- [X] This process follows the Milestone Framework as discussed in the Pocket Network Corporation's `Organization Handbook`  
+- [ ] This process `DOES NOT` follow the Milestone Framework as discussed in the Pocket Network Corporation's `Organization Handbook`  
+  
+If `DOES NOT` follow the Handbook's Milestone Framework, please describe in detail why below:  
+## Original Proposal  
+[Tx indexer Proposal](https://github.com/pokt-network/organization/blob/main/proposals/1-08-01-2021/tx-indexer-milestone-proposal.md)
+
+## Milestone Detailed Summary
+The current tx indexer implemented by Tendermint team offers subpar performance by forcing us to load all index keys sort them and then build a page.
+
+this default tx indexer has turned out to be inneficcient due to: 
+- Missindexing of txs: block events get appended to all txs
+- Inneficient retrieval of data: originally tendermint loaded all tx's in memory which was causing memory overflows, we proceeded to change that however due to limitations on the implementation we still must iterate through all tx's within range sort them and then build the page.
+
+The tx indexer should be able to understand the indexes it's going through and be able to build a page without having to load all keys and sorting them.
+
+I propose we keep this format in order to keep compatibility with indexing (following format will be broken into a regular expression)
+
+```regex
+([a-z .]+)\/([a-z 0-9]+)\/([0-9]+)\/([0-9]+)[\/]?
+```
+this makes sure we get any format like this
+```
+send.action/<address>/<height>/<index>
+claim.action/<address>/<height>/<index>
+tx.block/<height>/<height>/<index>
```

Comment:
What about indexing by transaction hash? Isn't that an important aspect of replay protection?

Reply:
Thas is correct, we can treat it as a branch case and just query the hash straight up.
this would look like this 

```
tx.hash=<hash>
```

and gives us O(1) for accessing the tx provided the hash.